### PR TITLE
Market search

### PR DESCRIPTION
**Pull request by Jesse193**
This adds functionality to the market search form to search by an address and a radius. This makes a get request to the BE to get an index of markets that display there name and each name can link to it's show page where a get request is made to the BE to display the market details such as its name, address, snap option, accepted payments and a description of where it is located. Searching by location automatically obtained by geocode is a work in progress

### CODE REVIEW

**Review by kbergstrom78 — APPROVED** [review]

File: app/controllers/markets_controller.rb [hunk 1]
```diff
@@ -1,7 +1,22 @@
 class MarketsController < ApplicationController
+  def search
+  end
+
   def index
+    if params[:use_my_location] != "0"
```
Comment (kaileykaes) [inline app/controllers/markets_controller.rb:6]: I believe lines 6-8 here will need to change when we learn how to use browser api location.

File: app/controllers/markets_controller.rb [hunk 2]
```diff
@@ -1,7 +1,22 @@
...
+      location = Geocoder.search(params[:use_my_location])
+    end
+
+    location = Geocoder.search(params[:search])
```
Comment (kaileykaes) [inline app/controllers/markets_controller.rb:10]: High potential for refactor into services, facades and poros for maintainability in the future.

File: app/controllers/markets_controller.rb [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+    params[:latitude] = location.first.coordinates[0]
+    params[:longitude] = location.first.coordinates[1]
+    params[:radius] = params[:radius].to_f
+    response = Faraday.get("https://foodhaven-be.onrender.com/api/v1/markets", params)
+    @markets = JSON.parse(response.body, symbolize_names: true)[:data]
   end
 
-  def search
+  def show
+    response = Faraday.get("https://foodhaven-be.onrender.com/api/v1/markets/#{params[:id]}")
```
Comment (kaileykaes) [inline app/controllers/markets_controller.rb:19]: Also great candidate for facade service poro refactor for future us to take care of.

File: spec/features/markets/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe "Market show page" do
+  it "can display link to show page" do
+    json_response = File.read('spec/fixtures/markets.json')
+    latitude = 30.69035
```
Comment (kaileykaes) [inline spec/features/markets/show_spec.rb:6]: Lines 6 - 9 look extraneous here. Will you take them out, please?

File: spec/features/markets/search_spec.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  it "can link to find markets page" do
+    visit root_path
+    click_on "Find a market near you"
+    expect(current_path).to eq(markets_search_path)
+    expect(page).to have_content("Enter address")
+    expect(page).to have_content("Within")
+  end
+  it "can search for market by address" do
+    json_response = File.read('spec/fixtures/markets.json')
+    latitude = 30.69035
```
Comment (kaileykaes) [inline spec/features/markets/search_spec.rb:13]: I don't think this is actually testing for a market search by address. It looks like this is simply getting an index of markets, but the params are not being passed anywhere in the request. Am I missing something?

File: app/services/market_service.rb [hunk 6]
```diff
@@ -1,14 +0,0 @@
-class MarketService
```
Comment (kaileykaes) [inline app/services/market_service.rb:1]: I think we'll want to refactor and bring this back for maintainability's sake in the future.

**Review by kaileykaes — CHANGES_REQUESTED** [review]

**Review by kaileykaes — APPROVED** [review]

**Pull request merged**